packages/Search/src/details/DTK_DetailsBatchedQueries.hpp

```diff
@@ -0,0 +1,188 @@
+                                                               n_queries );
+        Kokkos::parallel_for(
+            DTK_MARK_REGION( "assign_morton_codes_to_queries" ),
+            Kokkos::RangePolicy<ExecutionSpace>( 0, n_queries ),
+            KOKKOS_LAMBDA( int i ) {
+                Point xyz = Details::return_centroid( queries( i )._geometry );
+                for ( int d = 0; d < 3; ++d )
+                {
+                    double const a = scene_bounding_box.minCorner()[d];
+                    double const b = scene_bounding_box.maxCorner()[d];
+                    xyz[d] = ( a != b ? ( xyz[d] - a ) / ( b - a ) : 0 );
+                }
+                morton_codes( i ) = TreeConstruction<DeviceType>::morton3D(
+                    xyz[0], xyz[1], xyz[2] );
+            } );
+        Kokkos::fence();
+
+        Kokkos::View<int *, DeviceType> permute( "permute", n_queries );
+        iota( permute );
+        TreeConstruction<DeviceType>::sortObjects( morton_codes, permute );
+
+        return permute;
+    }
+
+    template <typename T>
+    static Kokkos::View<T *, DeviceType>
+    applyPermutation( Kokkos::View<int const *, DeviceType> permute,
+                      Kokkos::View<T *, DeviceType> v )
+    {
+        auto const n = permute.extent( 0 );
+        DTK_REQUIRE( v.extent( 0 ) == n );
+
+        decltype( v ) w( v.label(), n );
+        Kokkos::parallel_for(
+            DTK_MARK_REGION( "permute_entries" ),
+            Kokkos::RangePolicy<ExecutionSpace>( 0, n ),
+            KOKKOS_LAMBDA( int i ) { w( i ) = v( permute( i ) ); } );
+        Kokkos::fence();
+
+        return w;
+    }
+
+    static Kokkos::View<int *, DeviceType>
+    permuteOffset( Kokkos::View<int const *, DeviceType> permute,
+                   Kokkos::View<int const *, DeviceType> offset )
+    {
+        auto const n = permute.extent( 0 );
+        DTK_REQUIRE( offset.extent( 0 ) == n + 1 );
+
+        Kokkos::View<int *, DeviceType> tmp_offset( offset.label(), n + 1 );
+        Kokkos::parallel_for(
+            DTK_MARK_REGION( "adjacent_difference_and_permutation" ),
+            Kokkos::RangePolicy<ExecutionSpace>( 0, n ),
+            KOKKOS_LAMBDA( int i ) {
+                tmp_offset( permute( i ) ) = offset( i + 1 ) - offset( i );
+            } );
+        Kokkos::fence();
+
+        exclusivePrefixSum( tmp_offset );
+
+        return tmp_offset;
+    }
+
+    template <typename T>
+    static Kokkos::View<T *, DeviceType>
+    permuteIndices( Kokkos::View<int const *, DeviceType> permute,
+                    Kokkos::View<T const *, DeviceType> indices,
+                    Kokkos::View<int const *, DeviceType> offset,
+                    Kokkos::View<int const *, DeviceType> tmp_offset )
+    {
+        auto const n = permute.extent( 0 );
+
+        DTK_REQUIRE( offset.extent( 0 ) == n + 1 );
+        DTK_REQUIRE( tmp_offset.extent( 0 ) == n + 1 );
+        DTK_REQUIRE( lastElement( offset ) == indices.extent_int( 0 ) );
```

Comment:
Be consistent (at least inside one function), either use `extent` or `extent_int` but don't mix them

Reply:
Ok, will fix.